modules/calc/src/main/java/com/opengamma/strata/calc/runner/CalculationTasks.java

```diff
@@ -192,72 +152,152 @@ private CalculationTasks(List<CalculationTask> calculationTasks, List<Column> co
                 columnCount));
       }
     }
+
     // pull out the targets from the tasks
-    int targetCount = taskCount / columnCount;
-    this.targets = IntStream.range(0, targetCount)
-        .mapToObj(i -> calculationTasks.get(i * columnCount).getTarget())
-        .collect(toImmutableList());
+    int targetCount = (int) taskCount / columnCount;
+    CalculationTarget[] targets = new CalculationTarget[targetCount];
+    for (CalculationTask task : tasks) {
+      int rowIndex = task.getRowIndex();
+      if (targets[rowIndex] == null) {
+        targets[rowIndex] = task.getTarget();
+      } else if (targets[rowIndex] != task.getTarget()) {
+        throw new IllegalArgumentException("Tasks define two different targets for the same row");
```

Comment:
Probably worth including both targets in the exception message for easier debugging.